### PR TITLE
Added support for import and export of youtube playlists

### PR DESCRIPTION
**Pull request by dandimrod**
It has been bugging me that when you export and import user data it doesn't include any youtube playlists that the user is subscribed to, so every time I switch instances, I have to add my playlists by hand.

Unfortunately, I'm not a crystal expert and I don't have a working environment to test this, so some help on that regard would be appreciated.

I've added the title to the export for easier identification in case someone wants to remove some from the JSON by editing the file directly, but it can be removed either way, some input on this is also appreciated.

### CODE REVIEW

**Comment by dandimrod** [comment]
I think this would close https://github.com/iv-org/invidious/issues/2281 but I'm not sure since the description of that issue is quite sparse.

**Review by SamantazFox — COMMENTED** [review]
Code looks good to me.
However, we're ending up with the same concerns as here:
 * https://github.com/iv-org/invidious/pull/3973#issuecomment-1633271008
 * https://github.com/iv-org/invidious/pull/3968#discussion_r1253508149

**Comment by SamantazFox** [comment]
In regard to the issue mentioned above, can you remove the title and associated export logic?

**Comment by dandimrod** [comment]
@SamantazFox Sorry for the delay, I've changed the code to prevent use of `get_playlist`

**Review by SamantazFox — APPROVED** [review]
Code looks good :)
Can you switch your PR to "ready for review" please?

**Comment by SamantazFox** [comment]
Please fix the errors reported on by the CI build!

**Comment by DOOMMARINE117** [comment]
so is it possible for it to e fixed? works for importing subscriptions. 

what about a fix for playlists and watch history importing?

File: src/invidious/user/imports.cr [hunk 1]
```diff
@@ -101,6 +101,15 @@ struct Invidious::User
           end
         end
       end
+
+      if youtube_playlists = data["youtube_playlists"]?.try &.as_a?
+        playlists.each do |item|
```
Comment (RadoslavL) [inline src/invidious/user/imports.cr:106]: I think you meant to type `youtube_playlists.each do |item|`.

**Pull request closed without merging**